backend/src/utils.rs

```diff
@@ -118,43 +120,69 @@ where
             move |versioned_game| {
                 let mut g = InteractiveGame::new_from_state(versioned_game.game);
                 let mut associated_websockets = versioned_game.associated_websockets;
-                let mut msgs = operation(
+                let msgs_to_broadcast = operation(
                     &mut g,
                     versioned_game.monotonic_id,
                     &mut associated_websockets,
                 )
                 .map_err(EitherError::E2)?;
-                let game = g.into_state();
-                msgs.push(GameMessage::State {
-                    state: game.clone(),
-                });
+                let final_game_state = g.into_state();
                 Ok((
                     VersionedGame {
                         room_name: versioned_game.room_name,
-                        game,
+                        game: final_game_state,
                         associated_websockets,
                         monotonic_id: versioned_game.monotonic_id + 1,
                     },
-                    msgs,
+                    msgs_to_broadcast,
                 ))
             },
         )
         .await;
+
     match res {
-        Ok(_) => true,
-        Err(EitherError::E(_)) => {
-            let err = GameMessage::Error(format!("Failed to {action_description}"));
+        Ok(new_version) => match backend_storage.clone().get(room_name_.clone()).await {
+            Ok(updated_versioned_game) => {
+                if updated_versioned_game.monotonic_id == new_version {
+                    let targeted_state_msg = GameMessage::State {
+                        state: updated_versioned_game.game,
+                    };
+                    let _ = backend_storage
+                        .publish_to_single_subscriber(room_name_, ws_id, targeted_state_msg)
+                        .await;
+                    Ok(())
+                } else {
+                    let err_msg = format!("Operation succeeded but version mismatch after fetching state (expected {}, got {})", new_version, updated_versioned_game.monotonic_id);
```

Comment:
So, I think it's a correct/valid state as part of the protocol for a new version to come back with a higher monotonic ID than the one we tried to write - that means that we raced with another change, but the current state is still overall correct. I'm not sure that we should error in such a case.

It's also I think ~fine for the state in this process to fall out of date because any writes require the latest monotonic ID - do we need the optimistic read?